cmd/security-agent/subcommands/start/command.go

```diff
@@ -89,10 +90,12 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 				forwarder.Bundle,
 				fx.Provide(defaultforwarder.NewParamsWithResolvers),
 				demultiplexer.Module,
+				orchestratorForwarderImpl.Module,
+				fx.Supply(orchestratorForwarderImpl.Params{UseOrchestratorForwarder: false}),
 				fx.Provide(func() demultiplexer.Params {
 					opts := aggregator.DefaultAgentDemultiplexerOptions()
 					opts.UseEventPlatformForwarder = false
-					opts.UseOrchestratorForwarder = false
```

Comment:
Indeed, it doesn't and this is the reason why it is not enabled. See   `fx.Supply(orchestratorForwarderImpl.Params{UseOrchestratorForwarder: false})` few [lines upper](https://github.com/DataDog/datadog-agent/pull/21058/files/9320f52fffd3618cac5cb5f7eb5c945c1a95d4c8#diff-dddc20fd992d763e4375be1091d58113660de54d3124172c473626183297ec21R94).

Reply:
Because the `demultiplexer` uses the `orchestrator forwarder`, `orchestrator forwarder` is always mandatory when using the `demultiplexer` (Fx refuses to start the application if the module and the params are missing).
Note: There are units tests that check subcommands provides all the require modules and params. There is [a linter](https://github.com/DataDog/datadog-agent/blob/7.49.x/tasks/components.py#L324) to make sure all subcommand are tested (I also have a card to check there is no missing tests for other cases).

Reply:
Ok understood